importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -6032,30 +6083,78 @@ void MusicXMLParserNotations::addToScore(ChordRest* const cr, Note* const note,
+                              _logger->logError(QString("unknown fermata type %1").arg(direction), &_e);
+                              }
+                        addFermataToChord(cr, annotation.symId(), direction);
+                        }
+                  else {
+                        if (annotation.name() == "strong-accent") {
+                              if (direction != "" && direction != "up" && direction != "down") {
+                                    direction = (const char *) 0;
+                                    _logger->logError(QString("unknown %1 type %2").arg(annotation.name()).arg(direction), &_e);
+                                    }
+                              }
+                        else if (annotation.name() == "harmonic" || annotation.name() == "delayed-turn"
+                                 || annotation.name() == "turn" || annotation.name() == "inverted-turn") {
+                              if (annotation.name() == "delayed-turn") {
+                                    // TODO: actually this should be offset a bit to the right
+                                    }
+                              if (placement != "above" && placement != "below") {
+                                    placement = (const char *) 0;
+                                    _logger->logError(QString("unknown %1 placement %2").arg(annotation.name()).arg(placement), &_e);
+                                    }
+                              }
+                        else {
+                              direction = (const char *) 0; // TODO: Check for other symbols that have direction/type
+                              placement = (const char *) 0; // TODO: Check for other symbols that have placement
+                              }
+                        addArticulationToChord(cr, annotation.symId(), direction, placement);
+                        }
+                  }
+            else if (annotation.parent() == "notations") {
+                  if (annotation.name() == "slur") {
+                        addSlur(annotation, slurs, cr, tick, _logger, &_e);
+                        }
+                  else if (note && (annotation.name() == "glissando" || annotation.name() == "slide")) {
+                        addGlissandoSlide(annotation, note, glissandi, spanners, _logger, &_e);
+                        }
+                  else if (note && annotation.name() == "tied") {
+                        addTie(_score, note, cr->track(),
+                               annotation.attribute("type"), annotation.attribute("orientation"), annotation.attribute("line-type"),
+                               tie, _logger, &_e);
+                        }
+                  }
+            else if (annotation.parent() == "ornaments") {
+                  if (annotation.name() == "mordent" || annotation.name() == "inverted-mordent") {
+                        addMordentToChord(cr, annotation.name(),
```

Comment:
In retrospect it may be better to pass annotations as a const& to functions like addMordentToChord.